Add CLI tool with commands for key generation, address validation, transaction decoding, script analysis, and block parsing

JAGADISHSUNILPEDNEKAR opened the pull request:
# Add Bitcoin Utils CLI Tool

This PR adds a **simple yet powerful command-line interface** to the python-bitcoin-utils library, making core Bitcoin functionality **easily accessible** from the command line without requiring users to write Python code.

### Features Implemented

- **Address Validation**: Validate P2PKH, P2SH, and P2WPKH addresses
- **Key Pair Generation**: Create new key pairs or derive from existing WIF keys
- **Transaction Decoding**: Parse raw Bitcoin transactions into human-readable JSON
- **Script Analysis**: Analyze Bitcoin scripts and identify common script types
- **Block Parsing**: Extract information from raw Bitcoin blocks

### Benefits

- Makes the library **accessible to non-programmers** and command-line users
- Provides **educational tools** for understanding Bitcoin's inner workings
- **Simplifies common Bitcoin tasks** for developers and power users
- Demonstrates library functionality through **practical examples**

### Implementation Notes

- Built using Python's **argparse** for a clean, standard CLI experience
- **Compatible with all existing library functionality**
- Includes **comprehensive test coverage**
- Follows the library's API patterns and conventions

### Usage Examples

```bash
# Generate a Bitcoin keypair
bitcoin-utils generate

# Validate a Bitcoin address
bitcoin-utils validate 1BgGZ9tcN4rm9KBzDn7KprQz87SZ26SAMH --type p2pkh --pubkey 0450863AD64A87AE8A2FE83C1AF1A8403CB53F53E486D8511DAD8A04887E5B23522CD470243453A299FA9E77237716103ABC11A1DF38855ED6F2EE187E9C582BA6

# Decode a raw transaction
bitcoin-utils decode <transaction_hex>

# Analyze a Bitcoin script
bitcoin-utils script <script_hex>

# Parse a Bitcoin block
bitcoin-utils block <block_file_path>
```

This CLI tool **increases the utility** of the python-bitcoin-utils library **without changing any of its core functionality**.

<img width="1440" alt="Screenshot 2025-03-16 at 2 14 39 AM" src="https://github.com/user-attachments/assets/ebf73535-374e-47bb-8afd-f57b94ecc87e" />

JAGADISHSUNILPEDNEKAR committed:
Add CLI tool with commands for key generation, address validation, transaction decoding, script analysis, and block parsing

karask commented:
Hi @JAGADISHSUNILPEDNEKAR !

Please name the executable: 'bu' 

And add some instructions on how make it available as an executable in the top comments of bitcoin_utils_cli.py   (later on we might move it to the README file).

Thanks for the contribution. Not sure if it was in the TODO list I have but it was definitely something I had in mind to do!

JAGADISHSUNILPEDNEKAR committed:
Rename executable to 'bu' and add installation instructions

JAGADISHSUNILPEDNEKAR commented:
Thank you @karask for the feedback! I've made the requested changes:

1. Renamed the executable from "bitcoin-utils" to "bu" in all documentation references
2. Added detailed installation instructions in the top comments of `bitcoin_utils_cli.py` showing how to:
   - Make the file executable with `chmod +x`
   - Create a symlink in the user's PATH
   - An alternative pip installation method if included in a package
3. Updated usage examples to use "bu" instead of "bitcoin-utils"

I've kept the actual filename as `bitcoin_utils_cli.py` for clarity in the codebase, but the end-user will access it through the shorter "bu" command as requested.

The changes are minimal and focused only on the naming and installation instructions as requested, with all functionality preserved.

Let me know if you need any changes.

Thanks again
